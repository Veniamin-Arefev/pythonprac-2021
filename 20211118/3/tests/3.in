import random


def timer(func):
    import time
    def timed(*args, **kw):
        start = time.time()
        result = func(*args, **kw)
        te = time.time()
        print(f"{func.__name__} worked for {time.time() - start:0.5f} seconds")
        return result

    return timed


alp = Alpha(c=10, z=2, a=42)
alq = AlphaQ(c=10, z=2, a=42)


@timer
def work(alpha, count, seed=42):
    random.seed(seed)
    for i in range(count):
        setattr(alpha, random.choice(ascii_lowercase), random.randint(228, 1337 + 100500))


iter = 10000
work(alp, iter)
work(alq, iter)
