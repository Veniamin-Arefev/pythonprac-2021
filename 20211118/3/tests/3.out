work worked for 0.01200 seconds
work worked for 0.01100 seconds
